src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -707,17 +740,30 @@ private void StartCollection()
         /// </summary>
         private void SingleStep(bool forceGC)
         {
+            // Testing purposes only.
```

Comment:
do you think these if statements and construction of a new function within each singleStep could affect performance?

what about using `#if DEBUG`?

Reply:
I agree, that way it is also clearer that these statements are needed for debugging purposes.

Reply:
are cicd tests run on debug builds ?

Reply:
no, they use release builds.

Reply:
Just so that we can run it in release builds, I could duplicate the FullGC method for testing purposes ...